pkg/discovery/discovery.go

```diff
@@ -252,6 +254,16 @@ func (d *tidbDiscovery) VerifyPDEndpoint(pdURL string) (string, error) {
 		return pdURL, err
 	}
 
+	// if local pd doesn't exist, return target cluster pd peer addr
+	if tc.Heterogeneous() && tc.WithoutLocalPD() {
```

Comment:
If it's not remote heterogeneous cluster, no need to use the PD peer address, so we'd better check remote here.